### PR TITLE
feat(gadget-sdk): improve `MultiJobRunner` builder

### LINKED ISSUE #371: [TASK] Follow the traditional builder pattern in `MultiJobRunner`
# Overview

Right now, the `MultiJobRunner` kind of acts as a nested builder, in that it itself is a builder, but also has the `with_job()` method, which then spins off another builder.

That makes it a little difficult to read:

```rust
#[gadget_sdk::main(env)]
async fn main() {
    let x_square = blueprint::XsquareEventHandler {
        service_id: env.service_id.unwrap(),
        client: client.clone(),
        signer,
    };

    let x_square2 = blueprint::XsquareEventHandler {
        service_id: env.service_id.unwrap(),
        client: client.clone(),
        signer,
    };

    MultiJobRunner::new(&env)
        .with_job()
        .with_default_price_targets()
        .finish(x_square)
        .with_job()
        .with_default_price_targets()
        .finish(x_square2)
        .run()
        .await?;
}
```

It'd be better to instead:

* Keep `MultiJobRunner` simple. You want to add a job, use `MultiJobRunner::job()`.
* Remove the `with_` prefixes, builders typically use, for example, `price_targets(PriceTargets)` rather than `with_price_targets(PriceTargets)`.
* Keep defaults as default. When creating a `JobBuilder`, it should implicitly have the default price targets, rather than calling `with_default_price_targets()`.

Ideally, the above example would look like:

```rust
#[gadget_sdk::main(env)]
async fn main() {
    let x_square = blueprint::XsquareEventHandler {
        service_id: env.service_id.unwrap(),
        client: client.clone(),
        signer,
    };

    let x_square2 = blueprint::XsquareEventHandler {
        service_id: env.service_id.unwrap(),
        client: client.clone(),
        signer,
    };

    MultiJobRunner::new()
        .job(JobBuilder::new(&env).job(x_square).build())
        .job(JobBuilder::new(&env)
                .price_targets(/* Something, if you want */)
                .job(x_square2)
                .build())
        .run()
        .await?;
}
```

### PR DESCRIPTION
**Pull request by Serial-ATA**
Jobs can be built using tuples now. Some comparisons:

### Old

```rust
MultiJobRunner::new(&env)
    .with_job()
    .with_default_price_targets()
    .finish(x_square)
```

### New

```rust
MultiJobRunner::new(&env)
    .job(x_square)
```

### Old

```rust
use tangle_subxt::tangle_testnet_runtime::api::runtime_types::tangle_primitives::services::PriceTargets;

let price_targets = PriceTargets {
    cpu: 5,
    mem: 10,
    storage_hdd: 15,
    storage_ssd: 20,
    storage_nvme: 25
};

MultiJobRunner::new(&env)
    .with_job()
    .with_price_targets(price_targets)
    .finish(x_square)
```

### New

```rust
use tangle_subxt::tangle_testnet_runtime::api::runtime_types::tangle_primitives::services::PriceTargets;

let price_targets = PriceTargets {
    cpu: 5,
    mem: 10,
    storage_hdd: 15,
    storage_ssd: 20,
    storage_nvme: 25
};

MultiJobRunner::new(env)
    .job((x_square2, price_targets))
```

closes #371

### CODE REVIEW

File: sdk/src/job_runner.rs [hunk 1]
```diff
@@ -7,38 +7,105 @@ use std::future::Future;
...
-            >,
-        >,
-    >,
->;
+/// Wrapper for `tangle_subxt`'s [`PriceTargets`]
+///
+/// This provides a [`Default`] impl for a zeroed-out [`PriceTargets`].
+///
+/// [`PriceTargets`]: tangle_subxt::tangle_testnet_runtime::api::runtime_types::tangle_primitives::services::PriceTargets
+pub struct PriceTargets(TangleSubxtPriceTargets);
```
Comment (shekohex) [inline sdk/src/job_runner.rs:28]: We can provide a default implementation in subxt and remove this wrapper. But it is good for now.

**Review by shekohex — APPROVED** [review]
Looks good.

**Comment by tbraun96** [comment]
Just my `$0.02`. The tuple notation is awkward to me simply because I haven't really seen it in builder notation often and isn't intuitive.

How about a compromise:

```rust
let price_targets = PriceTargets {
    cpu: 5,
    mem: 10,
    storage_hdd: 15,
    storage_ssd: 20,
    storage_nvme: 25
};

MultiJobRunner::new(env)
    .job(x_square)
    .with_price_targets(price_targets)
    .job(x_square2)
    .job(x_square3)
    .with_default_price_targets()
    .job(s_square4)
    [...]
    .run()
    .await?
```

**Comment by Serial-ATA** [comment]
@tbraun96 the tuples are just for convenience, they actually get converted to `JobBuilder`s.

```rust
let price_targets = PriceTargets {
    cpu: 5,
    mem: 10,
    storage_hdd: 15,
    storage_ssd: 20,
    storage_nvme: 25
};

// These two are equivalent

MultiJobRunner::new(env)
    .job((x_square2, price_targets))

MultiJobRunner::new(env)
    .job(JobBuilder::new(x_square2).price_targets(price_targets))

// As are these

MultiJobRunner::new(env)
    .job(x_square)

MultiJobRunner::new(env)
    .job(JobBuilder::new(x_square))
```

**Comment by tbraun96** [comment]
> @tbraun96 the tuples are just for convenience, they actually get converted to `JobBuilder`s.
> 
> ```rust
> let price_targets = PriceTargets {
>     cpu: 5,
>     mem: 10,
>     storage_hdd: 15,
>     storage_ssd: 20,
>     storage_nvme: 25
> };
> 
> // These two are equivalent
> 
> MultiJobRunner::new(env)
>     .job((x_square2, price_targets))
> 
> MultiJobRunner::new(env)
>     .job(JobBuilder::new(x_square2).price_targets(price_targets))
> 
> // As are these
> 
> MultiJobRunner::new(env)
>     .job(x_square)
> 
> MultiJobRunner::new(env)
>     .job(JobBuilder::new(x_square))
> ```

Oh cool

**Comment by drewstone** [comment]
@tbraun96 I'm still surprised how you think this makes sense

> MultiJobRunner::new(env)
>     .job(x_square)
>     .with_price_targets(price_targets)
>     .job(x_square2)
>     .job(x_square3)
>     .with_default_price_targets()
>     .job(s_square4)
>     [...]
>     .run()
>     .await?
> ```

Where are the price targets for the jobs? Is it the price target after it's specified? What about `x_square2` would it be default? This is confusing lol. The tupleness is still 100% clearer on what is being attributed to what.

**Comment by drewstone** [comment]
@Serial-ATA @Tjemmmic we should meet or you two should to discuss ensuring the Eigenlayer runner is also made compatible with this system.

**Pull request merged**